grc/core/Platform.py

```diff
@@ -158,7 +158,13 @@ def build_block_library(self):
 
         # Add blocks to block tree
         for key, block in self.blocks.iteritems():
-            category = self._block_categories.get(key, block.category)
+            category = block.category
+            if not len(category):
+                category = self._block_categories.get(key, block.category)
+            elif category != self._block_categories.get(key, category):
```

Comment:
This doesn't let block tree info override block local categories!
May I suggest to leave the original (single line) piece of code and to the warning check with
`if catergory and category != block.category:`